packages/amplify-category-function/src/provider-utils/awscloudformation/utils/layerParams.ts

```diff
@@ -194,56 +195,40 @@ class LayerVersionState implements LayerVersionMetadata {
   }
 
   setPermissions(permissions: LayerPermission[]) {
-    this.permissions = [];
-    permissions.forEach(permission => {
-      if (permission.type === Permission.public) {
-        const permissionPublic: PublicLayer = {
-          type: Permission.public,
-        };
-        this.permissions.push(permissionPublic);
-      }
-      if (permission.type === Permission.private) {
-        const permissionPrivate: PrivateLayer = {
-          type: Permission.private,
-        };
-
-        this.permissions.push(permissionPrivate);
-      }
-      if (permission.type === Permission.awsOrg) {
-        const orgsPermissions: OrgsLayer = {
-          type: Permission.awsOrg,
-          orgs: permission.orgs,
-        };
-        this.permissions.push(orgsPermissions);
-      }
-      if (permission.type === Permission.awsAccounts) {
-        const accountsPermissions: AccountsLayer = {
-          type: Permission.awsAccounts,
-          accounts: permission.accounts,
-        };
-        this.permissions.push(accountsPermissions);
+    this.permissions = permissions.map(permission => {
+      switch (permission.type) {
+        case Permission.public:
+          return {
+            type: Permission.public,
+          } as PublicLayer;
+        case Permission.private:
+          return {
+            type: Permission.private,
+          };
+        case Permission.awsOrg:
+          return {
+            type: Permission.awsOrg,
+            orgs: permission.orgs,
+          };
+        case Permission.awsAccounts:
+          return {
+            type: Permission.awsAccounts,
+            accounts: permission.accounts,
+          };
```

Comment:
nit: return type definitions should be consistent

Reply:
These are LayerPermission types and TS is smart enough to figure out the union